autoload/lsp/utils.vim

```diff
@@ -138,3 +138,14 @@ function! lsp#utils#error(msg) abort
     echom a:msg
     echohl NONE
 endfunction
+
+function! lsp#utils#echo_with_truncation(msg) abort
+    let l:msg = a:msg
+    let l:winwidth = winwidth(0)
```

Comment:
Note that there is more space for an "echo" than this when there are vertical splits. `&columns` is a closer approximation to the amount of room to echo.